code/modules/mining/equipment/wormhole_jaunter.dm

```diff
@@ -89,3 +89,91 @@
 			if(ishuman(L))
 				shake_camera(L, 20, 1)
 				addtimer(CALLBACK(L, /mob/living/carbon.proc/vomit), 20)
+
+/obj/item/wormhole_jaunter/contractor
+	name = "emergency extraction flare"
+	icon = 'icons/obj/lighting.dmi'
+	desc = "A single-use extraction flare that will let you create a portal to any beacon on the station. You must choose the destination beforehand, else it will target a random beacon. The portal is generated 5 seconds after activation, and has 1 use."
+	icon_state = "flare-contractor"
+	item_state = "flare"
+	var/destination
+
+/obj/item/wormhole_jaunter/contractor/verb/set_destination()
+	set category = "Object"
+	set name = "Change Portal Destination"
+	set src in usr
+
+	var/list/L = list()
+	var/list/areaindex = list()
+
+	for(var/obj/item/radio/beacon/R in GLOB.beacons)
+		var/turf/T = get_turf(R)
+		if(!T)
+			continue
+		if(!is_teleport_allowed(T.z))
+			continue
+		var/tmpname = T.loc.name
+		if(areaindex[tmpname])
+			tmpname = "[tmpname] ([++areaindex[tmpname]])"
+		else
+			areaindex[tmpname] = 1
+		L[tmpname] = R
+
+	var/desc = input("Please select a location to target.", "Flare Target Interface") in L
+	destination = L[desc]
+	return
+
+/obj/item/wormhole_jaunter/contractor/attack_self(mob/user) // message is later down
+	activate(user, TRUE)
+
+/obj/item/wormhole_jaunter/contractor/activate(mob/user)
+	if(!turf_check(user))
+		return
+	if(!destination)
+		var/list/L = get_destinations(user)
+		if(!L.len)
+			to_chat(user, "<span class='notice'>[src] found no beacons in the sector to target.</span>")
+			return
+		destination = pick(L)
+	var/obj/effect/temp_visual/getaway_flare/F = new(get_turf(src))
+	user.visible_message("<span class='notice'>[user] pulls out a black and gold flare and lights it.</span>",\
+						  "<span class='notice'>You light an emergency extraction flare, initiating the extraction process.</span>")
+	user.drop_item()
+	forceMove(F)
+	addtimer(CALLBACK(src, .proc/create_portal, destination), 5 SECONDS)
+
+/obj/item/wormhole_jaunter/contractor/proc/create_portal(turf/destination)
+	new /obj/effect/portal/redspace/getaway(get_turf(src), get_turf(destination), src, 100)
+	qdel(src)
+
+/obj/item/wormhole_jaunter/contractor/emag_act(mob/user)
+	to_chat(user, "<span class='notice'>Emaging [src] has no effect.</span>")
+
+/obj/item/wormhole_jaunter/contractor/chasm_react(mob/user)
+	return //This is not an instant getaway portal like the jaunter
+
+/obj/item/storage/box/syndie_kit/escape_flare
+	name = "emergency extraction kit"
+
+/obj/item/storage/box/syndie_kit/escape_flare/New()
+	..()
+	new /obj/item/wormhole_jaunter/contractor(src)
+	new /obj/item/radio/beacon/emagged(src)
+	return
+/obj/effect/portal/redspace/getaway
```

Comment:
```suggestion

/obj/effect/portal/redspace/getaway
```

Reply:
Instead of making an entire new subtype, just instance it above and edit the portal.
```dm
var/obj/effect/portal/P = new /obj/effect/portal/redspace/getaway(get_turf(src), get_turf(destination), src, 100)
P.one_use = TRUE
```